ndbench-es-plugins/src/main/java/com/netflix/ndbench/plugin/es/ConstantStepWiseRateIncreaser.java

```diff
@@ -0,0 +1,150 @@
+        }
+        if (!(finalRate > 0)) {
+            throw new IllegalArgumentException("finalRate must be > 0");
+        }
+        if (!(finalRate > initRate)) {
+            throw new IllegalArgumentException("finalRate must be > initRate");
+        }
+        if (!(rampPeriodMillisecs > 0)) {
+            throw new IllegalArgumentException("rampPeriodMillisecs must be > 0");
+        }
+        if (!(incrementIntervalMillisecs > 0)) {
+            throw new IllegalArgumentException("incrementIntervalMillisecs must be > 0");
+        }
+        if (rampPeriodMillisecs % incrementIntervalMillisecs != 0) {
+            throw new IllegalArgumentException(
+                    "rampPeriodMillisecs should be evenly divisible by incrementIntervalMillisecs");
+        }
+        if (rampPeriodMillisecs / incrementIntervalMillisecs > MAX_STEPS) {
+            throw new IllegalArgumentException(
+                    "rampPeriodMillisecs / incrementIntervalMillisecs should not exceed MAX_STEPS (" + MAX_STEPS + ")");
+        }
+
+
+        final ArrayList<Cell> tmpClockTimeToRateTable = new ArrayList<Cell>();
+
+        int numSteps = rampPeriodMillisecs / incrementIntervalMillisecs;
+        double spread = (finalRate - initRate) * 1.0;
+        double rateIncrementPerStep = spread / numSteps;
+
+        for (int i = 0; i <= numSteps; i++) {
+            tmpClockTimeToRateTable.add(
+                    new Cell(
+                            i * incrementIntervalMillisecs,
+                            initRate + (i * rateIncrementPerStep)
+                    )
+            );
+        }
+
+        // initialize high values sentinel cell
+        tmpClockTimeToRateTable.add(new Cell(Long.MAX_VALUE, 0 /* dummy never used for rate */));
+
+        indexOfLastInTable = numSteps + 1;
+        assert (tmpClockTimeToRateTable.size() == indexOfLastInTable + 1);
+        assert (tmpClockTimeToRateTable.get(numSteps).time == rampPeriodMillisecs);
+        assert (tmpClockTimeToRateTable.get(numSteps).rate == finalRate);
+
+
+        ImmutableList.Builder<Cell> cellBuilder = new ImmutableList.Builder<>();
+        clockTimeToRateTable = cellBuilder.addAll(tmpClockTimeToRateTable).build();
+    }
+
+    /**
+     * Build internal array mapping relative-from-zero clock time values to the rate that will be returned once system
+     * clock exceeds time value in a particular cell (but is less than the time value in the immediately
+     * succeeding cell);
+     * <p>
+     * Ignore the  possible race condition that arises if multiple threads call this method at around the same time
+     * when an instance is an uninitialized state. Two (or maybe more) arrays will be concurrently built, but the
+     * system clock time to rate mappings will be close enough that it won't really matter -- auto-tuning behavior
+     * will work as expected.
+     * <p>
+     * TODO -- put this comment somewhere else
```

Comment:
I thought you already moved that comment above...

Reply:
yeah.. comments were all out of date.. fixing